script.py

```diff
@@ -25,45 +25,45 @@ def generate_token(client_id, client_secret):
 
     if(token_response.status_code == 200):
         return token_response.json()['access_token']
-    else:
-        print('Invalid credentials')
 
 
-def request_vehicles(*arg):
-    if len(arg) == 1:
-        return requests.get(f'{osu_API}v1/beaverbus/vehicles', headers={
-            'Authorization': f'Bearer {arg[0]}'
-        })
-    if len(arg) == 2:
-        return requests.get(f'{osu_API}v1/beaverbus/vehicles/{arg[1]}',
-                            headers={'Authorization': f'Bearer {arg[0]}'}
+def request_vehicles(id):
+    authorization_header = {'Authorization': f'Bearer {token}'}
+    beaverbus_vehicles = 'v1/beaverbus/vehicles/'
+    if id:
+        return requests.get(f'{osu_API}{beaverbus_vehicles}{id}',
```

Comment:
You can just simply store `{osu_API}{beaverbus_vehicles}` as one variable within this funciton.

Reply:
Your current method is pretty good, but one further improvement, this function can be improved like this:

```python
def request_vehicles(id):
    authorization_header = {'Authorization': f'Bearer {token}'}
    vehicles_endpoint = 'v1/beaverbus/vehicles/'
    url = f'{vehicles_endpoint}/{id}' if id else vehicles_endpoint

    return requests.get(url, headers=authorization_header)
```

Try to think about it and see if you can utilize this to improve your other codes?